### PR TITLE
850 create adpp

### LINKED ISSUE #850: Create ADPP

### PR DESCRIPTION
**Pull request by RealSlimMahdi**
Close #850 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://roche-gsk.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://roche-gsk.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::documents()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://roche-gsk.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by RealSlimMahdi** [comment]
Check failing because of missing PP object in data().

**Comment by RealSlimMahdi** [comment]
#850

**Comment by thomas-neitmann** [comment]
Thanks a lot @RealSlimMahdi! We'll merge once the PP dataset makes it to {admiral.test}'s main branch.

File: NEWS.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ are not recorded in the data. (#589)
 
 - `derive_var_ontrtfl()` now has a `new_var` parameter to support the derivation of `ONTRxxFL` and `ONTRTwFL` variables (#721)
 
+- New dataset template `ad_adpp.R` which creates Pharmacokinetics Parameters Analysis Dataset
```
Comment (thomas-neitmann) [inline NEWS.md:20]: Please include the issue number here.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: R/data.R [hunk 2]
```diff
@@ -65,3 +65,12 @@
 #' Derived from the `adsl` and `vs` datasets using `{admiral}` (\url{https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_advs.R})
 #'
 "admiral_advs"
+
+#' Pharmacokinetics Parameters Analysis Dataset
+#'
+#' An example vital signs analysis dataset
+#'
+#' @source
+#' Derived from the [adsl] and [pp] datasets using `{admiral}` (\url{https://github.com/Roche-GSK/admiral/blob/main/inst/templates/ad_adpp.R})
```
Comment (thomas-neitmann) [inline R/data.R:74]: Please remove the links here and put the dataset name in backticks.

File: R/data.R [hunk 3]
```diff
@@ -65,3 +65,12 @@
...
+#'
+"adpp"
```
Comment (thomas-neitmann) [inline R/data.R:76]: GIven our new naming convention this should be `admiral_adpp`.

File: inst/templates/ad_adpp.R [hunk 4]
```diff
@@ -0,0 +1,151 @@
+# Name: ADPP
+#
+# Label: Pharmacokinetics Parameters Analysis Dataset
+#
+# Input: pp, adsl
+library(admiral)
+library(admiral.test) # Contains example datasets from the CDISC pilot project
```
Comment (thomas-neitmann) [inline inst/templates/ad_adpp.R:7]: This should be `library(admrialtest)` after we renamed the package.

File: inst/templates/ad_adpp.R [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+# ---- Load source datasets ----
+
+# Use e.g. haven::read_sas to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+# Load PP and Adsl
+data("pp")
+data("adsl")
```
Comment (thomas-neitmann) [inline inst/templates/ad_adpp.R:21]: Please replace with

```
data("admiral_pp")
data("admiral_adsl")
```

File: inst/templates/ad_adpp.R [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    by = c("STUDYID", "USUBJID")
+  )
+
+# Final Steps, Select final variables and Add labels
+# This process will be based on your metadata, no example given for this reason
+# ...
+
+# ---- Save output ----
+
+save(adpp, file = "data/adpp.rda", compress = "bzip2")
```
Comment (thomas-neitmann) [inline inst/templates/ad_adpp.R:151]: Please change to `save(adpp, file = "data/admiral_adpp.rda", compress = "bzip2")`.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks a lot @RealSlimMahdi! I merged the latest changes from `devel` into your branch. There's a couple of minor things to change since we changed the naming convention of the datasets to include the `admiral_` prefix.

**Comment by bms63** [comment]
Do you think this can be finalized today?  We can probably give a day or two, if needed, as this would be a great addition to our toolbox?

File: inst/templates/ad_adpp.R (hunk 6 above)
Comment (RealSlimMahdi) [reply inst/templates/ad_adpp.R]: ok

File: R/data.R (hunk 3 above)
Comment (RealSlimMahdi) [reply R/data.R]: ok but I still have an issue here when I try to generate the doc

**Comment by RealSlimMahdi** [comment]
@bms63 SOrry I tried to update. But each time I go back to the repo I get more errors.
This time I get:
CleanShot 2022-05-16 at 19 25 12@2x
NOt sure whats going on

**Comment by bms63** [comment]
ah yes i am familar with some of those errors.  

try to do a `renv::restore()` from your console to get the latest version we are all using.   Hopefully, that will resolve knirtr warning 

How large is the admiral_adpp object?  We might have some issues with data storage in the package.  I had to drop the admiral_adlb dataset until we figure this out #1133 

The export piece needs in your script:

```
admrail_adpp <- adpp

save(admiral_adpp, file = "data/admiral_adpp.rda", compress = "bzip2")
```

this took me a minute to figure out, but the rda file name is different then the object name.  


Since this is a one time load of admiral data I would just change it back to:

```
dir <- tempdir() # Change to whichever directory you want to save the dataset in
save(adpp, file = file.path(dir, "adpp.rda"), compress = "bzip2")
```

so a user can just grab the code.

**Comment by bms63** [comment]
ACtually I think the knitr warning can be ignored.

**Comment by bms63** [comment]
Some discussion on fixing the knitr warning in this Pull Request https://github.com/pharmaverse/admiral/pull/1082

**Review by thomas-neitmann — APPROVED** [review]

**Comment by thomas-neitmann** [comment]
Thanks a lot for your support @RealSlimMahdi 👏 🥳

**Comment by RealSlimMahdi** [comment]
Thanks, @thomas-neitmann & @bms63 :)

**Comment by rossfarrugia** [comment]
awesome @RealSlimMahdi - thank you!! i should have mentioned at the admiral community meeting last week that we also now cover ADPP - this is another great addition for admiral.

**Comment by RealSlimMahdi** [comment]
Yes, hopefully, ADPC and ADNCA will be a bit less painful. I need to stop rebasing and start merging lol

**Pull request merged**